drivers/keyboard.c

```diff
@@ -0,0 +1,93 @@
+/* ------------------------------------------------------------------------ *\
+**
+**  This file is part of the Chaos Kernel, and is made available under
+**  the terms of the GNU General Public License version 2.
+**
+**  Copyright (C) 2017 - Benjamin Grange <benjamin.grange@epitech.eu>
+**
+\* ------------------------------------------------------------------------ */
+
+#include <kernel/init.h>
+#include <kernel/interrupts.h>
+#include <arch/x86/asm.h>
+#include <drivers/keyboard.h>
+#include <stdio.h>
+#include <arch/x86/apic.h>
+
+static char const fr_azerty_charset[128] =
+{
+	0,
+	[16] = 'a',
+	[17] = 'z',
+	[18] = 'e',
+	[19] = 'r',
+	[20] = 't',
+	[21] = 'y',
+	[22] = 'u',
+	[23] = 'i',
+	[24] = 'o',
+	[25] = 'p',
+	[30] = 'q',
+	[31] = 's',
+	[32] = 'd',
+	[33] = 'f',
+	[34] = 'g',
+	[35] = 'h',
+	[36] = 'j',
+	[37] = 'k',
+	[38] = 'l',
+	[39] = 'm',
+	[44] = 'w',
+	[45] = 'x',
+	[46] = 'c',
+	[47] = 'v',
+	[48] = 'b',
+	[49] = 'n',
+	[28] = '\n',
+	[14] = '\b',
+	[57] = ' ',
+};
+
+static volatile char input_buffer[PAGE_SIZE] = { 0 };
```

Comment:
These variables doesn't need to be `volatile`.

Reply:
done.